Content.Shared/Pinpointer/PinpointerComponent.cs

```diff
@@ -11,52 +12,89 @@ namespace Content.Shared.Pinpointer;
 [Access(typeof(SharedPinpointerSystem))]
 public sealed partial class PinpointerComponent : Component
 {
-    // TODO: Type serializer oh god
-    [DataField("component"), ViewVariables(VVAccess.ReadWrite)]
-    public string? Component;
+    /// <summary>
+    ///     A list of components that will be searched for when selected from the verb menu.
+    ///     The closest entities found with one of the components in this list will be added to the StoredTargets.
+    /// </summary>
+    [DataField]
+    public ComponentRegistry Components = new();
+
```

Comment:
Is this supposed to be marked as required.

Reply:
It can be empty if you want a pinpointer that can't search for any components, but can store targets through any other methods.